contracts/vortex/CarbonVortex.sol

```diff
@@ -631,6 +631,10 @@ contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable,
         if (sourceAmount == 0) {
             revert InvalidTrade();
         }
+        // revert if unnecessary native token is received
+        if (_targetToken != NATIVE_TOKEN && msg.value > 0) {
```

Comment:
You can refactor the `!=` and `==` check into a single flow, but it's not critical